### PR TITLE
FIPS checksum script sometimes left leading whitespace on lines

### LINKED ISSUE #15562: FIPS filter is not ignoring whitespace correctly.
#15539 was comment only changes and the fips checker showed differences

It should be ignoring whitespace.

The FIPS check-summing doesn't handle spaces perfectly.

Specificaly, these two lines are considered different:
```c
    statement;  /* comment */
    statement;
```
So adding or removing an end of line comment is detected as significant.

Refer to #15539 in pmeth_lib.c

### PR DESCRIPTION
**Pull request by paulidale**
The conditions are a bit convoluted requiring end of line white space on the preceding line and leading white space on the line in question.  This most easily occurs with an end of line comment since we're fairly good at not having end of line white space.

Fixes #15562

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: util/lang-compress.pl [hunk 1]
```diff
@@ -104,6 +96,10 @@
         }
         defined $1 ? $1 : " "
     }gsxe;
+
+    # Clean up spaces at start and end of lines
+    s/^ //mg;
+    s/ $//mg;
```
Comment (levitte) [inline util/lang-compress.pl:102]: I would actually use `\h+` instead of a mere space, just in case.

Comment (paulidale) [reply util/lang-compress.pl:102]: Okay, the previous substitution is changing \h+ to space so it's kind of moot....

Comment (levitte) [reply util/lang-compress.pl:102]: Oh, right!

so, never mind...

(note to self: coffee first, comment later)

Comment (paulidale) [reply util/lang-compress.pl:102]: Cool, I've reverted the change.

**Review by levitte — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master with updated checksums.

**Pull request closed without merging**